types/basetypes/float64_value.go

```diff
@@ -108,7 +132,12 @@ func (f Float64Value) Equal(other attr.Value) bool {
 		return true
 	}
 
-	return f.value == o.value
+	// Not possible to create a known Float64Value with a nil value, but check anyways
+	if f.value == nil || o.value == nil {
+		return false
```

Comment:
I'm not sure if its better or worse to treat both values being `nil` as `true` or not -- feel free to discard if you disagree. But this should still also prevent the potential panic.

```suggestion
	if f.value == nil {
		return o.value == nil
```

Reply:
I was unsure as well, but I think if I have to choose, I'd rather do what you proposed. ⚡